NOAA-OWP/ras2fim
# [1pt] PR: new get unit inputs tool

This tool, named `s3_get_unit_inputs.py`,  can take in a HUC8 at a minimum, scan the local driver to see exactly files ras2fim.py needs for input files and goes get the files from S3.  For some files that are HUC specific like inputs\dems\ras_3dep_HUC8_10m\, it will pull down only the file that applies to the given HUC.  It can process only one HUC at a time (per script run).  You only need to edit the `s3_unit_download_files.lst` (and likely check it in), if you want to change the what needs to be downloaded generally for ras2fim.py.

This file simplifies getting setup for a ras2fim.py run and will also be helpful with automation.

The files to be downloaded are based on a config file named s3_unit_download_files.lst for quick updates or testing.

Closes Issue [290](https://github.com/NOAA-OWP/ras2fim/issues/290).

### Additions  

- `config`
    - `s3_unit_download_files.lst`: The changeable list of which files need to be downloaded for ras2fim.py run.
- `tools`
    - `s3_get_unit_inputs.py`: As described above.

### Changes  
- `src`
    - `clip_dem_from_shape.py`: variable name change.
    - `ras2fim.py`: variable name change.
    - `shared_variables.py`:  one variable name change plus a few new one ones for this tool.
- `tools`
    - `acquire_and_preprocess_3dep_dems.py`: variable name change.
    - `run_unit_benchmark_tests.py`: Some linting, updated some notes and text.
    - `s3_shared_functions.py`: minor text adjustments

### Testing

To test:
The basic min arg command is: `python ./tools/s3_get_unit_inputs.py -w {some single huc number}

1) Looking at the `s3_unit_download_files.lst`, delete some or all of them on your local file system, then run the command. It should fill in the wholes. Use a huc number that you do not already have a dem for in inputs/dems/ras_3dep_HUC8_10m.
2) Run it again as it and it should tell you they all exist.

Feel free to test any other args if you like. I already have.


### Checklist

_You may update this checklist before and/or after creating the PR. If you're unsure about any of them, please ask, we're here to help! These items are what we are going to look for before merging your code._

- [x] Informative and human-readable title, using the format: `[_pt] PR: <description>`
- [x] Pre-commit executed and linting changes made.
- [x] Links are provided if this PR resolves an issue, or depends on another other PR
- [x] If submitting a PR to the `dev` branch (the default branch), you have a descriptive [Feature Branch](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow) name using the format: `dev-<description-of-change>` (e.g.: `dev-revise-levee-masking`)
- [x] The feature branch you're submitting as a PR is up to date (merged) with the latest `dev` branch
- [x] Changes adhere to [PEP-8 Style Guidelines](https://peps.python.org/pep-0008/)
- [ ] n/a - Placeholder code is flagged / future todos are captured in comments
- [x] Project documentation has been updated ([CHANGELOG](/doc/CHANGELOG.md) and/or [README](/README.md))
- [x] [Reviewers requested](https://help.github.com/articles/requesting-a-pull-request-review/)

### Merge Checklist (For Technical Lead use only)

- [ ] Update [CHANGELOG](/docs/CHANGELOG.md) with latest version number and merge date
